onnxruntime/core/providers/cuda/rnn/cudnn_rnn_base.cc

```diff
@@ -288,12 +317,20 @@ Status CudnnRnnBase<T>::ComputeInternal(OpKernelContext* ctx) const {
     }
   }
 
-  if (sequence_lens_data != nullptr && y_h_data != nullptr && y_data != nullptr) {
+  if ((CUDNN_RNN_RELU == rnn_mode_ || CUDNN_RNN_TANH == rnn_mode_) && sequence_lens_data != nullptr && y_h_data != nullptr && y_data != nullptr) {
+    auto count = sequence_lens->Shape().Size();
+    int device_id = 0;
```

Comment:
>device_id [](start = 8, length = 9)

You can get the device id from provider_. This won't work with multiple GPU

Reply:
I can fix it in this PR. and will create a work item to setup test environment and add test cases for multiple GPUs.

---
In reply to: [292754893](https://github.com/microsoft/onnxruntime/pull/899#discussion_r292754893) [](ancestors = 292754893)

Reply:
Thanks! LGTM

---
In reply to: [292757715](https://github.com/microsoft/onnxruntime/pull/899#discussion_r292757715) [](ancestors = 292757715,292754893)